menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala

```diff
@@ -57,16 +63,35 @@ class SchemaController @Autowired() (
     val fileContent = new String(file.getBytes)
 
     val sparkStruct = origFormat match {
-      case Some("struct") | Some("") | None => sparkMenasConvertor.convertAnyToStructType(fileContent)
-      case Some("copybook") =>
-        val parsedSchema = CopybookParser.parseTree(fileContent)
-        val cobolSchema = new CobolSchema(parsedSchema, SchemaRetentionPolicy.CollapseRoot, false)
-        cobolSchema.getSparkSchema
-      case Some(x) => throw new UnsupportedOperationException(s"$x is not a recognized schema format. Menas currently supports: 'struct' and 'copybook'")
+      case Some(SchemaTypeStruct) | Some("") | None =>
+        try {
+          sparkMenasConvertor.convertAnyToStructType(fileContent)
+        } catch {
+          case NonFatal(e) => throw SchemaParsingException(SchemaTypeStruct, e.getMessage)
```

Comment:
```suggestion
          case NonFatal(e) => throw SchemaParsingException(SchemaTypeStruct, e.getMessage, e)
```
Let's add the cause here.